### PR TITLE
Initial updates to global_cycle for Noah-MP land model

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 
The global_cycle program was updated to recognize and process Noah-MP restart files. Currently, only the greenness fraction is updated for Noah-MP. Other required fields for Noah-MP will be added later after consultation with the land team. The global workflow group would like something to test with.

Changes are backwards compatible when processing Noah restart files.

## TESTS CONDUCTED: 
Documented below.

## DEPENDENCIES:
See https://github.com/NOAA-EMC/global-workflow/pull/1315

## DOCUMENTATION:
N/A

## ISSUE: 
Part of #761.

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
The branch at 4e724bb was checked out and compiled on Hera. The `global_cycle` regression tests were run. All tests failed.

When compared to the baseline files, none of the data records were different. Rather, some of the attributes were different. 

For example, there was a remark about a missing 'checksum':

```
DIFFER : VARIABLE "stc" IS MISSING ATTRIBUTE WITH NAME "checksum" IN FILE "/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/global_cycle/baseline_data/c768.fv3gfs/20190730.000000.sfcanl_data.tile6.nc"
DIFFER : NUMBER OF ATTRIBUTES : VARIABLE : stc : 3 <> 2
```

And the Time dimension was different:

```
DIFFER : VARIABLE : shdmax : DIMENSION Time IS RECORD IN FILE "20190730.000000.sfcanl_data.tile6.nc" BUT NOT IN "/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/global_cycle/baseline_data/c768.fv3gfs/20190730.000000.sfcanl_data.tile6.nc"
```

The baseline file had one time dimension:

```
dimensions:
        xaxis_1 = 768 ;
        yaxis_1 = 768 ;
        zaxis_1 = 4 ;
        Time = 1 ;
```

The file output from the branch version of `global_cycle` had an unlimited time dimension:

```
dimensions:
        xaxis_1 = 768 ;
        yaxis_1 = 768 ;
        zaxis_1 = 4 ;
        Time = UNLIMITED ; // (1 currently)
```

The difference in the time dimension is due to the branch updating a pre-existing restart file, which had an unlimited dimension. The baseline file was written by the head of develop, which writes an entirely new restart file with one time dimension.

None of these differences indicate a problem with the branch.

**Comment by GeorgeGayno-NOAA** [comment]
Additional differences between the branch and baseline files were noted in the "C768.landincsnow" and "C768.lndincsoil" tests. These tests don't run the surface cycling step or do an NSST update. Rather, they only update the snow/soil temperature records using increments from JEDI/GSI. The branch updated an existing restart file, which contained NSST records. Because NSST was not turned on, 'develop' wrote a new file without the NSST records (according to the logic of its write routine). 

```
DIFFER : VARIABLE : z_c : DOES NOT EXIST IN "/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/global_cycle/baseline_data/c768.lndincsnow/20190730.000000.sfcanl_data.tile6.nc"
```

The snow and soil temperature records were the same, however, as they should be.

**Comment by GeorgeGayno-NOAA** [comment]
The branch at 4e724bb was compiled on Dogwood, Jet and Orion. The regression tests failed in the same way.

**Review by aerorahul — APPROVED** [review]
I can't review the contents of the PR as they are beyond my expertise.
However, I have tested this branch in a cycled experiment within global-workflow.
A global-workflow PR with this branch has been opened at: 
https://github.com/NOAA-EMC/global-workflow/pull/1315

**Comment by GeorgeGayno-NOAA** [comment]
@HelinWei-NOAA, can you review?

**Comment by HelinWei-NOAA** [comment]
> @HelinWei-NOAA, can you review?

Sure.

File: sorc/global_cycle.fd/cycle.f90 [hunk 1]
```diff
@@ -712,14 +715,24 @@ SUBROUTINE SFCDRV(LUGB, IDIM,JDIM,LSM,LENSFC,LSOIL,DELTSFC,  &
...
-                 T2M,Q2M,VETFCS,SOTFCS,USTAR,FMM,FHH,        &
-                 SICFCS,SIHFCS,SITFCS,                       &
-                 TPRCP,SRFLAG,SNDFCS,                        &
-                 VMNFCS,VMXFCS,SLPFCS,ABSFCS,                &
-                 SLCFCS,SMCFCS,STCFCS,                       &
-                 IDIM,JDIM,LENSFC,LSOIL,DO_NSST,NSST)
+ IF (IS_NOAHMP) THEN
+
+   CALL WRITE_DATA(LENSFC,IDIM,JDIM,LSOIL,DO_NSST,NSST,VEGFCS=VEGFCS)
+
```
Comment (HelinWei-NOAA) [inline sorc/global_cycle.fd/cycle.f90:721]: @GeorgeGayno-NOAA If this cycle code is also used in DA, then the soil moisture(nudge) and snow depth (envelop strategy) will be also be updated with NoahMP. We had some discussions during our land meeting yesterday. So we need to have a tag-up meeting with you to make some decisions. For this PR, should we go ahead with the current version to have one working version?

**Review by HelinWei-NOAA — APPROVED** [review]

File: sorc/global_cycle.fd/cycle.f90 (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply sorc/global_cycle.fd/cycle.f90:721]: I spoke with @barlage briefly yesterday about a meeting next week to discuss which fields need to be updated for Noah-MP. However, @aerorahul wants a Noah-MP version of global_cycle he can start testing in the workflow right away, even if that version is only partially complete.

**Comment by GeorgeGayno-NOAA** [comment]
Will merge, then update the consistency test baselines on each machine.

**Pull request merged**